lib/methods/ifwapichecktask.cpp

```diff
@@ -0,0 +1,314 @@
+	const Dictionary::Ptr& resolvedMacros, bool useResolvedMacros)
+{
+	using namespace boost::asio;
+	using namespace boost::beast;
+	using namespace boost::beast::http;
+
+	REQUIRE_NOT_NULL(checkable);
+	REQUIRE_NOT_NULL(cr);
+
+	CheckCommand::Ptr command = CheckCommand::ExecuteOverride ? CheckCommand::ExecuteOverride : checkable->GetCheckCommand();
+
+	Host::Ptr host;
+	Service::Ptr service;
+	tie(host, service) = GetHostService(checkable);
+
+	MacroProcessor::ResolverList resolvers;
+
+	if (MacroResolver::OverrideMacros)
+		resolvers.emplace_back("override", MacroResolver::OverrideMacros);
+
+	if (service)
+		resolvers.emplace_back("service", service);
+	resolvers.emplace_back("host", host);
+	resolvers.emplace_back("command", command);
+	resolvers.emplace_back("icinga", IcingaApplication::GetInstance());
+
+	String psCommand = MacroProcessor::ResolveMacros("$ifw_api_command$", resolvers, checkable->GetLastCheckResult(),
+		nullptr, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	Dictionary::Ptr arguments = MacroProcessor::ResolveMacros("$ifw_api_arguments$", resolvers, checkable->GetLastCheckResult(),
+		nullptr, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	Array::Ptr ignoreArguments = MacroProcessor::ResolveMacros("$ifw_api_ignore_arguments$", resolvers, checkable->GetLastCheckResult(),
+		nullptr, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	String psHost = MacroProcessor::ResolveMacros("$ifw_api_host$", resolvers, checkable->GetLastCheckResult(),
+		nullptr, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	String psPort = MacroProcessor::ResolveMacros("$ifw_api_port$", resolvers, checkable->GetLastCheckResult(),
+		nullptr, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	Dictionary::Ptr params = new Dictionary();
+
+	if (arguments) {
+		auto ignore (ignoreArguments->ToSet<String>());
+		ObjectLock oLock (arguments);
+		Array::Ptr emptyCmd = new Array();
+
+		for (auto& kv : arguments) {
+			if (ignore.find(kv.first) != ignore.end()) {
+				continue;
+			}
+
+			/* MacroProcessor::ResolveArguments() converts
+			 *
+			 * [ "check_example" ]
+			 * and
+			 * {
+			 * 	 "-f" = { set_if = "$example_flag$" }
+			 * 	 "-a" = "$example_arg$"
+			 * }
+			 *
+			 * to
+			 *
+			 * [ "check_example", "-f", "-a", "X" ]
+			 *
+			 * but we need the args one-by-one like [ "-f" ] or [ "-a", "X" ].
+			 */
+			Array::Ptr arg = MacroProcessor::ResolveArguments(
+				emptyCmd, new Dictionary({{kv.first, kv.second}}), resolvers,
+				checkable->GetLastCheckResult(), resolvedMacros, useResolvedMacros
+			);
+
+			switch (arg ? arg->GetLength() : 0) {
+				case 0:
+					break;
+				case 1:
+					params->Set(arg->Get(0), true);
+					break;
+				case 2:
+					params->Set(arg->Get(0), arg->Get(1));
+					break;
+				default: {
+					auto k (arg->Get(0));
+
+					arg->Remove(0);
+					params->Set(k, arg);
+				}
+			}
```

Comment:
Short examples for the JSON in each case wouldn't hurt. I'm especially curious about the `default:` case: that would transform `["-x", "a", "b", "c"]` into `{"-x": ["a", "b", "c"]}`. This pretty much implies that all CheckCommands that use this case have to use `repeat_key = false` (which is **not** the default). Is this consistent with all existing PowerShell checks?

Reply:
repeat_key? In the JSON structute the PowerShell daemon expects? How? 🤯

Reply:
No, in `CheckCommand.arguments`. If you had `repeat_key = true` (default) there and mutliple values, wouldn't `MacroProcessor::ResolveArguments()` return something like `["-x", "a", "-x", "b", "-x", "c"]` which would then be transformed to `{"-x": ["a", "-x", "b", "-x", "c"]}`.